chrome/script.browserify.js

```diff
@@ -232,6 +236,47 @@ function getFaviconUrl(domain) {
   return null;
 }
 
+function launchURL() {
+  var what = this.what;
+  var entry = this.entry;
+  chrome.runtime.sendNativeMessage(
+    app,
+    { action: "get", entry: this.entry },
+    function(response) {
+      if (chrome.runtime.lastError) {
+        error = chrome.runtime.lastError.message;
+        m.redraw();
+      } else {
+        // get url from login path if not available in the host app response
+        if (!response.hasOwnProperty("url") || response.url.length == 0) {
+          var parts = entry.split(/\//).reverse();
+          for (var i in parts) {
+            var part = parts[i];
+            var info = Tldjs.parse(part);
+            if (info.isValid && info.tldExists && info.domain !== null && info.hostname === part) {
+              response.url = part;
+              break;
+            }
+          }
+        }
+        // if a url is present, then launch a new tab via the background script
+        if (response.hasOwnProperty("url") && response.url.length > 0) {
+          var url = response.url.match(/^([a-z]+:)?\/\//i) ? response.url : "http://" + response.url;
+          chrome.runtime.sendMessage({action: "launch", url: url, username: response.u, password: response.p});
+          window.close();
+        } else {
+          // no url available
+          if (!response.hasOwnProperty("url")) {
+            resetWithError("Your host application is too old - must be at least 2.0.14 for URL launch.");
```

Comment:
First brick in building #164, I like this 👍 

However, we need to adjust the message, because with added parsing of domain it's not consistent anymore. With the old host application I can open url for password entries like "personal/google.com" just fine, until one day I try to open url for the entry "personal/notes" and suddenly get the error "Your host application is too old". 

I think we should say something like this: 

Unable to detect the URL to open for this entry. If you defined it in the password file, update your host application to the latest version.

Reply:
Good point - done.